src/pyj/read_book/cfi.pyj

```diff
@@ -57,21 +57,6 @@ def window_scroll_pos(w):  # {{{
     return w.pageXOffset, w.pageYOffset
 # }}}
 
-def viewport_to_document(x, y, doc):  # {{{
-    doc = doc or window.document
-    while doc is not window.document:
-        # we are in a frame
-        frame = doc.defaultView.frameElement
-        rect = frame.getBoundingClientRect()
-        x += rect.left
-        y += rect.top
-        doc = frame.ownerDocument
-    wx, wy = window_scroll_pos(doc.defaultView)
-    x += wx
-    y += wy
-    return x, y
-# }}}
-
```

Comment:
The ScrollViewport fools the rest of the code into thinking that things progress left-to-right, in a positive direction, but in RTL mode, things go right-to-left and become negative.

The coordinates this will return in RTL mode will be negative - but we want to use the ScrollViewport to pretend everything is positive, so move this function to the viewport and have it negate the value of the X coordinate returned if in RTL mode.